src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisition.cs

```diff
@@ -641,6 +643,63 @@ public async Task<AuthenticationResult> GetAuthenticationResultForAppAsync(
             }
         }
 
+        private MergedOptions GetMergedOptions(string? authenticationScheme, TokenAcquisitionOptions? tokenAcquisitionOptions)
+        {
+            MergedOptions mergedOptions;
+
+            if (tokenAcquisitionOptions != null
+                && tokenAcquisitionOptions.ExtraParameters != null
+                && tokenAcquisitionOptions.ExtraParameters.TryGetValue("MicrosoftIdentityOptions", out object? identityOptions)
+                && identityOptions is MicrosoftEntraApplicationOptions microsoftEntraApplicationOptions)
+            {
+                MergedOptions parentMergedOptions = _tokenAcquisitionHost.GetOptions(authenticationScheme ?? tokenAcquisitionOptions?.AuthenticationOptionsName, out _);
+                mergedOptions = new MergedOptions()
```

Comment:
This is brittle - everytime we add a new option to MicrosoftEntraApplicationOptions  we have to update this code? Can this be moved to a MergedOptions ctor? `MergedOptions(MicrosoftEntraApplicationOptions  o)` ?

Reply:
We'll use the Copy code generator when productized:
https://github.com/[jmprieur/CopyFromMethodGenerator](https://github.com/jmprieur/CopyFromMethodGenerator)